### PR TITLE
Leave management UI and calculation

### PR DESCRIPTION
**Pull request by prasanthchaduvula**
### Figma
https://www.figma.com/file/4gdPDRPqGqQqPYsO3Qj8CG/Miru-App?node-id=16947%3A316009&mode=dev

### What
- User can view all the applied leaves with date, type of leave and number of hours.
- Users can also view the leave balance with net hours and days for each leave type.
- Admins/owners can view other team member's leaves and balances.

### Demo

https://github.com/saeloun/miru-web/assets/52771571/a319494b-f2bf-4bf8-ba0d-5b73807c173b

### CODE REVIEW

**Comment by apoorv1316** [comment]
@prasanthchaduvula Please add specs for service

File: app/javascript/src/components/LeaveManagement/Container/Table/TableHeader.tsx [hunk 1]
```diff
@@ -2,23 +2,14 @@ import React from "react";
...
 const TableHeader = () => (
   <thead>
-    <tr className="flex flex-row items-center py-5">
-      <th
-        className="flex w-1/3 cursor-pointer pr-8 text-left text-xs font-normal tracking-widest text-miru-dark-purple-600 lg:w-3/12"
-        scope="col"
-      >
+    <tr className="flex w-full justify-between">
+      <th className="flex text-left text-xs font-normal tracking-widest text-miru-dark-purple-600 lg:w-1/4">
         DATE
```
Comment (SiddharthShringi) [inline app/javascript/src/components/LeaveManagement/Container/Table/TableHeader.tsx:7]: We can start using translations on the front-end side as well. We can use react i18next for that.

File: app/javascript/src/components/LeaveManagement/Header.tsx [hunk 2]
```diff
@@ -1,27 +1,31 @@
...
+}) => (
+  <div className="m-4 flex items-center justify-between lg:mx-0">
+    <span className="hidden text-3xl font-bold text-miru-dark-purple-1000 lg:inline">
+      Leave Management
+    </span>
+    {/* <CustomYearPicker
         currentYear={currentYear}
         setCurrentYear={setCurrentYear}
         wrapperClassName="text-miru-han-purple-1000"
         yearClassName="text-miru-han-purple-1000"
```
Comment (SiddharthShringi) [inline app/javascript/src/components/LeaveManagement/Header.tsx:19]: Do we not need this component? If not we can remove it.

File: app/javascript/src/components/LeaveManagement/Container/Table/TableHeader.tsx (hunk 1 above)
Comment (prasanthchaduvula) [reply app/javascript/src/components/LeaveManagement/Container/Table/TableHeader.tsx:7]: Will implement with translation in phaser2

File: app/javascript/src/components/LeaveManagement/Header.tsx (hunk 2 above)
Comment (prasanthchaduvula) [reply app/javascript/src/components/LeaveManagement/Header.tsx:19]: We need in phase 2

**Review by apoorv1316 — APPROVED** [review]

**Pull request merged**